server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -244,7 +244,7 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
 
             CheckedConsumer<List<LeafReaderContext>, IOException> leafSorter = l -> {};
             // try to rewrite numeric or date sort to the optimized distanceFeatureQuery
-            if ((searchContext.sort() != null) && SYS_PROP_REWRITE_SORT) {
+            if ((searchContext.sort() != null) && (searchContext.size() > 0) && SYS_PROP_REWRITE_SORT) {
```

Comment:
What if 'size' is 0 but 'from' is not, can this case uses sort optimization?  if it can, changing this place to "from + size" may be much better. 
Feel free about this, just an advice :P

Reply:
@yyff thanks for pointing this out! It seems unusual to perform a search with `from: n, size: 0`, but I do agree it's more consistent to check 'num hits' (which is `from + size`) instead of just `size`. @mayya-sharipova what are your thoughts on adjusting this?

Reply:
@yyff @jtibshirani Thanks for commenting.  I have addressed this in https://github.com/elastic/elasticsearch/pull/57250